portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoValueLocalServiceImpl.java

```diff
@@ -463,8 +463,12 @@ public ExpandoValue addValue(
 			String columnName, long classPK, Object data)
 		throws PortalException {
 
-		ExpandoColumn column = _expandoColumnLocalService.getColumn(
-			companyId, className, tableName, columnName);
+		ExpandoTable table = _expandoTablePersistence.fetchByC_C_N(
+			companyId, _classNameLocalService.getClassNameId(className),
+			tableName);
+
+		ExpandoColumn column = _expandoColumnPersistence.fetchByT_N(
```

Comment:
@slnn this is not an equivalent change. "expandoColumnLocalService.getColumn()" has a null check for the ExpandoTable, but here you use the table to call "table.getTableId()" which could lead to NPE.

I did not check all your changes, please make sure you take care all similar cases.

Reply:
OK, got it